an initial cask with a manual install for SoapUI until a better cask …

loverde opened the pull request:
It's not much more than a link to download the current installer and instructions to run it, but it's better than nothing at this point.  I'm not familiar enough with cask yet to know if it would be possible to automate the running of the installer from the command line.

vitorgalvao commented:
It’s missing `uninstall`. There are [scripts](https://github.com/caskroom/homebrew-cask/tree/master/developer/bin) and [documentation](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#uninstall-stanza-details) geared for that.

However, this cask has been submitted a few times before and never accepted due to issues with the app (see https://github.com/caskroom/homebrew-cask/pull/10606). Are those solved?

loverde commented:
There isn't an uninstaller with the application, though you can just delete the /Applications directory so I added a step to do that.  Also I dug a bit more and found a way to drive the install from the CLI without requiring a manual installation.  My install steps skip the optional HermesJMS step because that is actually a separate installer and would be a better fit for a separate cask if someone wants that.

I added the changes to the pull request.  And this now installs and uninstalls cleanly on my El Capitan machine without any manual intervention.  So I believe the issues mentioned previously have been resolved.

vitorgalvao commented on Casks/soapui.rb:
```diff
@@ -0,0 +1,37 @@
+  version '5.2.1'
+  sha256 '773cb3aaa269e14ed34f06d931b6898e6a8ad16d9d08d67cc4935e137bfbe37c'
+
+  url 'http://cdn01.downloads.smartbear.com/soapui/5.2.1/SoapUI-5.2.1.dmg'
```
Needs a `  # smartbear.com is the official download host per the vendor homepage` comment above this line ([example](https://github.com/caskroom/homebrew-cask/blob/cb487ad405f74cb1f26b93e3f3e8412c82a22e34/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/7c010b42bd7ac679c1f002a0368b36bb3f495612/doc/CASK_LANGUAGE_REFERENCE.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

Also, change this to `url "http://cdn01.downloads.smartbear.com/soapui/#{version}/SoapUI-#{version}.dmg"` so it’s easier to update in the future.

loverde replied:
done

vitorgalvao commented on Casks/soapui.rb:
```diff
@@ -0,0 +1,37 @@
+
+  installer :script => 'SoapUI 5.2.1 Installer.app/Contents/MacOS/JavaApplicationStub',
+            :args => %w[-c],
+            :input => [
```
You can put this all in a single line, so it doesn’t take up this much space. Can you briefly explain it, though?

loverde replied:
What those input lines are doing is responding to installer questions with a set of defaults.  It might be easier to just paste the installer output so you can see what I mean.

So I will put it on a single line for now.  Another option would be to comment just what each input line is doing with a trailing comment.  Let me know if you'd prefer that.

~~~~
==> This will install SoapUI 5.2.1 on your computer.
==> OK [o, Enter], Cancel [c]
==> Please read the following License Agreement. You must accept the terms of this agreement before continuing with the installation.
==>
==> -----------------
==> SoapUI, copyright (C) 2004-2014 smartbear.com
==> Licensed under the EUPL, Version 1.1 or - as soon as they will be approved
==> by the European Commission - subsequent
==> versions of the EUPL (the "Licence");
==> You may not use this work except in compliance with the Licence.
==> SoapUI is distributed in the hope that it will be useful, but WITHOUT ANY
==> WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
==> FOR A PARTICULAR PURPOSE.
==> -----------------
==> European Union Public Licence V. 1.1
==> EUPL (C) the European Community 2007
==> This European Union Public Licence (the “EUPL”) applies to the Work or
==> Software (as defined below) which is provided under the terms of this
==> Licence. Any use of the Work, other than as authorised under this Licence is
==> prohibited (to the extent such use is covered by a right of the copyright
==> holder of the Work).
==> The Original Work is provided under the terms of this Licence when the
==> Licensor (as defined below) has placed the following notice immediately
==> following the copyright notice for the Original Work:
==> Licensed under the EUPL V.1.1
==> or has expressed by any other mean his willingness to license under the
==> EUPL.
==> [Enter]
==>
==> 1.	Definitions
==> In this Licence, the following terms have the following meaning:
==> - The Licence: this Licence.
==> - The Original Work or the Software: the software distributed and/or
==> communicated by the Licensor under this Licence, available as Source Code
==> and also as Executable Code as the case may be.
==> - Derivative Works: the works or software that could be created by the
==> Licensee, based upon the Original Work or modifications thereof. This
==> Licence does not define the extent of modification or dependence on the
==> Original Work required in order to classify a work as a Derivative Work;
==> this extent is determined by copyright law applicable in the country
==> mentioned in Article 15.
==> - The Work: the Original Work and/or its Derivative Works.
==> - The Source Code: the human-readable form of the Work which is the most
==> convenient for people to study and modify.
==> - The Executable Code: any code which has generally been compiled and which
==> is meant to be interpreted by a computer as a program.
==> - The Licensor: the natural or legal person that distributes and/or
==> communicates the Work under the Licence.
==> - Contributor(s): any natural or legal person who modifies the Work under
==> the Licence, or otherwise contributes to the creation of a Derivative Work.
==> - The Licensee or “You”: any natural or legal person who makes any usage of
==> [Enter]
==> the Software under the terms of the Licence.
==> - Distribution and/or Communication: any act of selling, giving, lending,
==> renting, distributing, communicating, transmitting, or otherwise making
==> available, on-line or off-line, copies of the Work or providing access to
==> its essential functionalities at the disposal of any other natural or legal
==> person.
==>
==> 2.	Scope of the rights granted by the Licence
==> The Licensor hereby grants You a world-wide, royalty-free, non-exclusive,
==> sub- licensable licence to do the following, for the duration of copyright
==> vested in the Original Work:
==> - use the Work in any circumstance and for all usage,
==> - reproduce the Work,
==> - modify the Original Work, and make Derivative Works based upon the Work,
==> - communicate to the public, including the right to make available or
==> display the Work or copies thereof to the public and perform publicly, as
==> the case may be, the Work,
==> - distribute the Work or copies thereof,
==> - lend and rent the Work or copies thereof,
==> - sub-license rights in the Work or copies thereof.
==> Those rights can be exercised on any media, supports and formats, whether
==> now known or later invented, as far as the applicable law permits so.
==> In the countries where moral rights apply, the Licensor waives his right to
==> [Enter]
==> exercise his moral right to the extent allowed by law in order to make
==> effective the licence of the economic rights here above listed.
==> The Licensor grants to the Licensee royalty-free, non exclusive usage rights
==> to any patents held by the Licensor, to the extent necessary to make use of
==> the rights granted on the Work under this Licence.
==>
==> 3.	Communication of the Source Code
==> The Licensor may provide the Work either in its Source Code form, or as
==> Executable Code. If the Work is provided as Executable Code, the Licensor
==> provides in addition a machine-readable copy of the Source Code of the Work
==> along with each copy of the Work that the Licensor distributes or indicates,
==> in a notice following the copyright notice attached to the Work, a
==> repository where the Source Code is easily and freely accessible for as long
==> as the Licensor continues to distribute and/or communicate the W ork.
==>
==> 4.	Limitations on copyright
==> Nothing in this Licence is intended to deprive the Licensee of the benefits
==> from any exception or limitation to the exclusive rights of the rights
==> owners in the Original Work or Software, of the exhaustion of those rights
==> or of other applicable limitations thereto.
==>
==> 5.	Obligations of the Licensee
==> The grant of the rights mentioned above is subject to some restrictions and
==> [Enter]
==> obligations imposed on the Licensee. Those obligations are the following:
==> Attribution right: the Licensee shall keep intact all copyright, patent or
==> trademarks notices and all notices that refer to the Licence and to the
==> disclaimer of warranties. The Licensee must include a copy of such notices
==> and a copy of the Licence with every copy of the Work he/she distributes
==> and/or communicates. The Licensee must cause any Derivative Work to carry
==> prominent notices stating that the Work has been modified and the date of
==> modification.
==> Copyleft clause: If the Licensee distributes and/or communicates copies of
==> the Original Works or Derivative Works based upon the Original Work, this
==> Distribution and/or Communication will be done under the terms of this
==> Licence or of a later version of this Licence unless the Original Work is
==> expressly distributed only under this version of the Licence. The Licensee
==> (becoming Licensor) cannot offer or impose any additional terms or
==> conditions on the Work or Derivative Work that alter or restrict the terms
==> of the Licence.
==> Compatibility clause: If the Licensee Distributes and/or Communicates
==> Derivative Works or copies thereof based upon both the Original Work and
==> another work licensed under a Compatible Licence, this Distribution and/or
==> Communication can be done under the terms of this Compatible Licence. For
==> the sake of this clause, “Compatible Licence” refers to the licences listed
==> in the appendix attached to this Licence. Should the Licensee’s obligations
==> under the Compatible Licence conflict with his/her obligations under this
==> [Enter]
==> Licence, the obligations of the Compatible Licence shall prevail.
==> Provision of Source Code: When distributing and/or communicating copies of
==> the Work, the Licensee will provide a machine-readable copy of the Source
==> Code or indicate a repository where this Source will be easily and freely
==> available for as long as the Licensee continues to distribute and/or
==> communicate the Work.
==> Legal Protection: This Licence does not grant permission to use the trade
==> names, trademarks, service marks, or names of the Licensor, except as
==> required for reasonable and customary use in describing the origin of the
==> Work and reproducing the content of the copyright notice.
==>
==> 6.	Chain of Authorship
==> The original Licensor warrants that the copyright in the Original Work
==> granted hereunder is owned by him/her or licensed to him/her and that he/she
==> has the power and authority to grant the Licence.
==> Each Contributor warrants that the copyright in the modifications he/she
==> brings to the Work are owned by him/her or licensed to him/her and that
==> he/she has the power and authority to grant the Licence.
==> Each time You accept the Licence, the original Licensor and subsequent
==> Contributors grant You a licence to their contributions to the Work, under
==> the terms of this Licence.
==>
==> 7.	Disclaimer of Warranty
==> [Enter]
==> The Work is a work in progress, which is continuously improved by numerous
==> contributors. It is not a finished work and may therefore contain defects or
==> “bugs” inherent to this type of software development.
==> For the above reason, the Work is provided under the Licence on an “as is”
==> basis and without warranties of any kind concerning the Work, including
==> without limitation merchantability, fitness for a particular purpose,
==> absence of defects or errors, accuracy, non-infringement of intellectual
==> property rights other than copyright as stated in Article 6 of this Licence.
==> This disclaimer of warranty is an essential part of the Licence and a
==> condition for the grant of any rights to the Work.
==>
==> 8.	Disclaimer of Liability
==> Except in the cases of wilful misconduct or damages directly caused to
==> natural persons, the Licensor will in no event be liable for any direct or
==> indirect, material or moral, damages of any kind, arising out of the Licence
==> or of the use of the Work, including without limitation, damages for loss of
==> goodwill, work stoppage, computer failure or malfunction, loss of data or
==> any commercial damage, even if the Licensor has been advised of the
==> possibility of such damage. However, the Licensor will be liable under
==> statutory product liability laws as far such laws apply to the Work.
==>
==> 9.	Additional agreements
==> While distributing the Original Work or Derivative Works, You may choose to
==> [Enter]
==> conclude an additional agreement to offer, and charge a fee for, acceptance
==> of support, warranty, indemnity, or other liability obligations and/or
==> services consistent with this Licence. However, in accepting such
==> obligations, You may act only on your own behalf and on your sole
==> responsibility, not on behalf of the original Licensor or any other
==> Contributor, and only if You agree to indemnify, defend, and hold each
==> Contributor harmless for any liability incurred by, or claims asserted
==> against such Contributor by the fact You have accepted any such warranty or
==> additional liability.
==>
==> 10.	Acceptance of the Licence
==> The provisions of this Licence can be accepted by clicking on an icon “I
==> agree” placed under the bottom of a window displaying the text of this
==> Licence or by affirming consent in any other similar way, in accordance with
==> the rules of applicable law. Clicking on that icon indicates your clear and
==> irrevocable acceptance of this Licence and all of its terms and conditions.
==> Similarly, you irrevocably accept this Licence and all of its terms and
==> conditions by exercising any rights granted to You by Article 2 of this
==> Licence, such as the use of the Work, the creation by You of a Derivative
==> Work or the Distribution and/or Communication by You of the Work or copies
==> thereof.
==>
==> 11.	Information to the public
==> [Enter]
==> In case of any Distribution and/or Communication of the Work by means of
==> electronic communication by You (for example, by offering to download the
==> Work from a remote location) the distribution channel or media (for example,
==> a website) must at least provide to the public the information requested by
==> the applicable law regarding the Licensor, the Licence and the way it may be
==> accessible, concluded, stored and reproduced by the Licensee.
==>
==> 12.	Termination of the Licence
==> The Licence and the rights granted hereunder will terminate automatically
==> upon any breach by the Licensee of the terms of the Licence.
==> Such a termination will not terminate the licences of any person who has
==> received the Work from the Licensee under the Licence, provided such persons
==> remain in full compliance with the Licence.
==>
==> 13.	Miscellaneous
==> Without prejudice of Article 9 above, the Licence represents the complete
==> agreement between the Parties as to the Work licensed hereunder.
==> If any provision of the Licence is invalid or unenforceable under applicable
==> law, this will not affect the validity or enforceability of the Licence as a
==> whole. Such provision will be construed and/or reformed so as necessary to
==> make it valid and enforceable.
==> The European Commission may publish other linguistic versions and/or new
==> versions of this Licence, so far this is required and reasonable, without
==> [Enter]
==> reducing the scope of the rights granted by the Licence. New versions of the
==> Licence will be published with a unique version number.
==> All linguistic versions of this Licence, approved by the European
==> Commission, have identical value. Parties can take advantage of the
==> linguistic version of their choice.
==>
==> 14.	Jurisdiction
==> Any litigation resulting from the interpretation of this License, arising
==> between the European Commission, as a Licensor, and any Licensee, will be
==> subject to the jurisdiction of the Court of Justice of the European
==> Communities, as laid down in article 238 of the Treaty establishing the
==> European Community.
==> Any litigation arising between Parties, other than the European Commission,
==> and resulting from the interpretation of this License, will be subject to
==> the exclusive jurisdiction of the competent court where the Licensor resides
==> or conducts its primary business.
==>
==> 15.	Applicable Law
==> This Licence shall be governed by the law of the European Union country
==> where the Licensor resides or has his registered office.
==> This licence shall be governed by the Belgian law if:
==> - a litigation arises between the European Commission, as a Licensor, and
==> any Licensee;
==> [Enter]
==> - the Licensor, other than the European Commission, has no residence or
==> registered office inside a European Union country.
==>
==> ===
==>
==> Appendix
==> “Compatible Licences” according to article 5 EUPL are:
==> - GNU General Public License (GNU GPL) v. 2 - Open Software License (OSL) v.
==> 2.1, v. 3.0
==> - Common Public License v. 1.0
==> - Eclipse Public License v. 1.0
==> - Cecill v. 2.0
==>
==> I accept the agreement
==> Yes [1], No [2]
==> Where should SoapUI 5.2.1 be installed?
==> [/Applications]
==> Which components should be installed?
==> *: SoapUI [*1]
==> 1: Source [*2]
==> 2: HermesJMS [*3]
==> 3: Tutorials [*4]
==> (To show the description of a component, please enter one of *1, *2, *3, *4)
==> Please enter a comma-separated list of the selected values or [Enter] for the default selection:
==> [2,3]
==> Create a desktop icon?
==> Yes [y, Enter], No [n]
==> Extracting files ...
  Home
==> Setup has finished installing SoapUI 5.2.1 on your computer.
==> View the release notes?
==> Yes [y, Enter], No [n]
==> Run SoapUI 5.2.1?
==> Yes [y, Enter], No [n]
==> Finishing installation ...
~~~~

vitorgalvao commented on Casks/soapui.rb:
```diff
@@ -0,0 +1,37 @@
+
+  # Not installing HermesJMS above, but if we did, add:
+  # '/Applications/HermesJMS'
+  uninstall :delete => [
```
Are you sure the only thing the installer does is move this app?

loverde replied:
You can see the output of the install4j installer script above.  The options I'm choosing are accepting the license agreement, choosing only to install SoapUI (not HermesJMS or tutorials or source), install to /Applications, don't create a desktop icon, don't view the release notes, and don't launch soapui after installing.

The above doesn't directly move the package to /Applications.  It runs the install4j installer which then moves the contents to the /Applications directory and unzips and reorganizes some of the contents within that directory.  You can see the difference if you compare:

~~~~
ls -laR /opt/homebrew-cask/Caskroom/soapui/5.2.1/SoapUI\ 5.2.1\ Installer.app
~~~~

with:
~~~~
ls -laR /Applications/SoapUI-5.2.1.app
~~~~

vitorgalvao commented on Casks/soapui.rb:
```diff
@@ -0,0 +1,37 @@
+              'n'
+            ]
+
+  # Not installing HermesJMS above, but if we did, add:
```
Please remove the comments. As you said, this would be better suited for another cask.

loverde replied:
done

loverde commented:
Incorporated the comments above into the cask.

jawshooah commented:
Casks with similar installers just use the `-q` flag to run the installer quietly, presumably using defaults for any values normally prompted from the user. Have you tried that?

loverde commented:
So the problem with -q is that it will run the installation with the default settings.  However, digging through the [install4j cli docs](http://resources.ej-technologies.com/install4j/help/doc/index.html) it seems that it's possible to pass variables to override the defaults on the command line as well.  And to get those values you can look at your installation's .install4j directory that it creates.

So taking the values from /Applications/SoapUI-5.2.1.app/Contents/java/app/.install4j/response.varfile I was able to create a version of the cask that is quiet and avoids the noise during the installation.

I added comments to this affect to the cask.

Also, I tried to search the installation directory and the web for information about an automated uninstall, but there doesn't seem to be one.  And [this post](http://forum.soapui.org/viewtopic.php?f=5&t=22342), albeit for an older release, seems to imply you just need to use the move to trash method.

vitorgalvao commented:
Thank you for the contribution. It needed some fixes, so they were made in a6a12ae826af97843891e092d97be0c41d20eb6b. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.
